backends/s3/s3.go

```diff
@@ -346,6 +346,27 @@ func New(ctx context.Context, opt Options) (*Backend, error) {
 	return b, nil
 }
 
+// NewReader satisfies ReaderStorage and provides an optimized, streamed way
+// to fetch a blob from S3 server.
+func (b *Backend) NewReader(ctx context.Context, name string) (io.ReadCloser, error) {
+	obj, err := b.client.GetObject(ctx, b.opt.Bucket, name, minio.GetObjectOptions{})
+	if err = convertMinioError(err); err != nil {
+		return nil, err
+	}
+	if obj == nil {
+		return nil, os.ErrNotExist
+	}
+
+	info, err := obj.Stat()
+	if err = convertMinioError(err); err != nil {
+		return nil, err
+	}
+	if info.Key != name {
+		return nil, os.ErrNotExist
+	}
+	return obj, nil
```

Comment:
Directly returning the minio object will not cause `setMarker` to be called on close. We probably also want to increment metrics when performing streamed reads and writes, if feasible.